### ISSUE TITLE
Support for Hyperv on windows 

### ISSUE THREAD

**Issue by floradu88** [docker/kitematic#1463]
Hello,

I am working on a fork on adding support for hyperv. I will probably do a pull request if thats ok?

Regards,
Radu

**Comment by FrenchBen** [comment]
That would be great - Thanks!

**Comment by floradu88** [comment]
Thanks, will try to create it tomorrow.

**Comment by floradu88** [comment]
Submitted pull request with changes: https://github.com/docker/kitematic/pull/1491

**Comment by FrenchBen** [comment]
Yup saw it, but haven't had time to build a windows VM to do a review. Hopefully I get a bit of time towards the end of the week. 
One question was regarding the VM creation in HyperV, but it looks like you're using the hyperv machine driver?

**Comment by floradu88** [comment]
Yes, I am trying to see if hyper-v is installed. If it is installed, then I create the machine with hyper-v. I am using the same command as for virtualbox, but with different arguments.

**Comment by pmario** [comment]
Hi, I managed to get my branch working. \o/. 

 - To check if Hyper-V exists, I'm using the same command as docker-machine
 - kitematic can run in user mode. no "run as admin needed" .. except "create" (atm)
   - user needs to be in group: Hyper-V-Administrator
 - creating the VM with docker-machine needs a workaround, because 
      - [docker machine has an issue](https://github.com/docker/machine/issues/2989) with a hard coded dependency. ... Some more testing is needed. 
      - [MS seems to have one too](https://support.microsoft.com/en-us/kb/2249906). .... more testing needed.
 - the switch name is still hard coded, but without spaces now: `DockerVirtualSwitch`. No spaces simplifies the elevated command a lot. ... 
 - The second possibility would be, to use the same mechanism as docker-machine. .. They use the first one they find and provide a config parameter to change it. 
    - I think we should go with this solution.  --->   What do you think?
 - IMO the switch should be created by the User. .. we should just check it one exists and point them to the right docs, how to create one. ... We don't know the users security requirements. So let them create it. 
 - I'll try to publish my version tomorrow.

**Comment by floradu88** [comment]
Looks good. I haven't have time to do anything in the past week. But will take a look at your branch.

**Comment by pmario** [comment]
Link to #1161, IMO it seems to be useful code but forgotten.

**Comment by pmario** [comment]
## Hard coded requirements

- User needs to be part of the `Hyper-V-Administrators` group.
   - User needs elevated rights only once: When VM is created. 
   - see: [comment above](https://github.com/docker/kitematic/issues/1463#issuecomment-195819642) point 3
- The virtual switch needs to be named `KitematicVirtualSwitch`

### Setup Kitematic VSwitch

- Open the `Hyper-V Manager` Desktop App
- If you need Internet Access, you'll probably need to use `External Network`!
- see [msdn Hyper-V on Windows](https://msdn.microsoft.com/en-us/virtualization/hyperv_on_windows/quick_start/walkthrough_virtual_switch) for more info.

  - ![03 kitematicvirtualswitch_](https://cloud.githubusercontent.com/assets/374655/13741596/7815671c-e9d7-11e5-82e7-8cb33ead2f50.gif)

### Add user to Hyper-V-Administrators group

- Open your file `explorer`

- right click `This PC`: Manage (You have to be Admin to do this !) 

  - ![01 right click this pc manage_](https://cloud.githubusercontent.com/assets/374655/13741831/adff26be-e9d8-11e5-9027-678c5c38c48d.gif)

- System Tools: Local Users and Groups: Groups

  - ![02 manage groups hyper-v-admins_](https://cloud.githubusercontent.com/assets/374655/13742163/a14165b6-e9da-11e5-8a5d-5c9428cb9fa9.gif)

**Comment by pmario** [comment]
I can't test it with VBox. So someone should definitely see if it still works :)
